src/vcs/git.rs

```diff
@@ -192,6 +206,103 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file
+                        .path()
+                        .ok_or(Error::GitDiff("couldn't retrieve file path"))?;
```

Comment:
If we did the above enum this would get embedded in the top-level `Error`

Reply:
Also, I think having the file path in the payload is useful for the error message. I don't want to be asking myself "which file path" in a few months time :smile:

Reply:
That's a bit of a conundrum, since the error is due to the path not being available!

Reply:
Ah hahaha silly me! Disregard my madness